### DISCUSSION TITLE
should-i-use-self-contained-or-framework-dependent-publishing-in-docker-images/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#346] [Announcements]
# should-i-use-self-contained-or-framework-dependent-publishing-in-docker-images/

In this post I compare the impact of the framework-dependent vs self-contained mode on Docker image size, taking layer caching into account

https://andrewlock.net/should-i-use-self-contained-or-framework-dependent-publishing-in-docker-images/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 15 2020, 05:11, in reply to [markmsmh](https://disqus.com/by/markmsmh/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/346#discussioncomment-6059726)</em>

---

<p>I realise that, but if you're <i>not</i> using trimming, there's 0 reason for using self-contained in Docker images right? All the down sides (no layer caching) with none of the benefits (potentially smaller overall footprint)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 16 2020, 09:22, in reply to [swisshero](https://disqus.com/by/SwissHero/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/346#discussioncomment-6059730)</em>

---

<p>No typo, I'm only talking about deploying using Docker, as opposed to deploying to a VM, to bare metal, or to App Service, for example</p>

[a comment by an automated account is not shown]

**Comment by mutkin** [comment]
A typo under Framework-Dependent pros:
Pro: you can path the .NET runtime for all your apps, without having to re-compile.
Should be 'patch', not 'path'.
